### ISSUE TITLE
create a PR template

### ISSUE THREAD

**Issue by khushal87** [jainaman224/Algo_Ds_Notes#657]
Can i start working on it ?? @jainaman224

**Comment by khushal87** [comment]
How about it ??

----------------------
Fixes # 


#### Checklist

- [] I have read the [Contribution & Best practices Guide of Algo_Ds_Notes ] and my PR follows them.
- [] My branch is up-to-date with the Upstream branch.
- [] I have added necessary documentation (if appropriate)


#### Changes proposed in this pull request:

-
-

**Comment by khushal87** [comment]
ping @jainaman224

**Comment by jainaman224** [comment]
Please think of improved version of this template if possible.

**Comment by jainaman224** [comment]
I appreciate the idea of PR template.

**Comment by khushal87** [comment]
okay

**Comment by helios1101** [comment]
Can i propose my template style if in case needed urgently ? Or should i wait for some time.

**Comment by jainaman224** [comment]
@helios1101 Not urgent. Let @khushal87 have a go.

**Comment by khushal87** [comment]
@jainaman224 what changes do you want me to implement in this PR .Do suggest me ..For now i kept it as simple as possible so that it would be easy for users too ?? what changes can i implement??

**Comment by jainaman224** [comment]
@helios1101 Can you also suggest your template.

**Comment by jainaman224** [comment]
Let's take best features from both templates.

**Comment by helios1101** [comment]
This PR reference issue #

### Branch Name: 
 This will directly make us know which folder he is working 
### Programming Files Added :
- [ ] Cpp
- [ ] C
- [ ] C#
- [ ] Python
  _This will help admin to put their handles after every checkbox of different languages like ._ 
- [ ] Cpp @jainaman224 

### I have read and by PR abide by :
- [ ] Code Of Conduct And Contributing.md

**Comment by khushal87** [comment]
So basically a language field can be merged too

**Comment by khushal87** [comment]
How about it ??

----------------------
Fixes # 


#### Checklist

- [] I have read the [Contribution & Best practices Guide of Algo_Ds_Notes ] and my PR follows them.
- [] My branch is up-to-date with the Upstream branch.
- [] I have added necessary documentation (if appropriate)


#### Changes proposed in this pull request:

-
-

Languages Used :--

-
-
-
-

**Comment by jainaman224** [comment]
Please remind me on Sunday.
Let's have a discussion.

**Comment by jainaman224** [comment]
Things are really important for us.

1. Sample input/output.
2. Some guidelines regarding file name and check box.
3.  Branch Name guidelines.

What are your thoughts as guidelines, upstream and documentation seems to be less useful.
For documentation we can say update the folder Readme. If there is no Readme add one with links.

**Comment by jainaman224** [comment]
Something around code convention and squashing will be a great plus.

**Comment by jainaman224** [comment]
Add some convention on commit message etc etc. These small things can impact drastically.
I need to your suggestions and a coordinated template. Both of you will get marks for it.

**Comment by khushal87** [comment]
> 
> 
> Things are really important for us.
> 
>     1. Sample input/output.
> 
>     2. Some guidelines regarding file name and check box.
> 
>     3. Branch Name guidelines.
> 
> 
> What are your thoughts as guidelines, upstream and documentation seems to be less useful.
> For documentation we can say update the folder Readme. If there is no Readme add one with links.

I think the name would suffice may it be of the code of the ds/algo or the documentation of it 
Input/Output is a nice idea though

**Comment by khushal87** [comment]
Like we can have a checkbox that says (I have followed the standard code convention and my code contains the necessary documentation ,comments,input and output).
This could direct the user to have it on there code before submitting

**Comment by jainaman224** [comment]
Great.

**Comment by khushal87** [comment]
okay. Let me make it and soon i will share it

**Comment by jainaman224** [comment]
@helios1101 Your feedback will be great.

**Comment by helios1101** [comment]
> > Things are really important for us.
> > ```
> > 1. Sample input/output.
> > 
> > 2. Some guidelines regarding file name and check box.
> > 
> > 3. Branch Name guidelines.
> > ```
> > What are your thoughts as guidelines, upstream and documentation seems to be less useful.
> > For documentation we can say update the folder Readme. If there is no Readme add one with links.
> 
> I think the name would suffice may it be of the code of the ds/algo or the documentation of it
> Input/Output is a nice idea though

Yeah, i too think that giving sample input output will be great. Also people sometimes uses logic which are nice but not enough to get the core concept, so a helpful documentation in support of explaination.

**Comment by helios1101** [comment]
Also, sometimes their are contribution of a particular priogramme file to existing folders. For eg. in Binary Tree folder person1 made 4 programme files, and person 2 made 2 files, so to distinguish such commits we need a checkbox as in support to folder or new folder.

**Comment by jainaman224** [comment]
Nice. I feel you two will come up with great template.

**Comment by jainaman224** [comment]
@helios1101 mention khushal in that feature template issue as well.
All 3 of us try to make good templates.

**Comment by khushal87** [comment]
How about it ??

----------------------
Fixes # 


#### Checklist

- [ ] I have read the [Contribution & Best practices Guide of Algo_Ds_Notes ] and my PR follows them.
- [ ] My branch is up-to-date with the Upstream branch.
- [ ] I have followed the standard code convention and my code contains the necessary documentation ,comments,input and output


#### Changes proposed in this pull request:

-
-

#### Languages Used :--

- [ ] C
- [ ] C++
- [ ] Java
- [ ] Python
- [ ] JavaScript
- [ ] Ruby
- [ ] Go
- [ ] Machine Learning Algorithm
- [ ] Others,please specify  below
--#Mention Other Languages

#### Files Added(specify there names)
 -
 -
 -

A template according to the suggestions till now ?? how is it ??

**Comment by helios1101** [comment]
> How about it ??
> 
> Fixes #
> 
> #### Checklist:
> * [ ]  I have read the [Contribution & Best practices Guide of Algo_Ds_Notes ] and my PR follows them.
> * [ ]  My branch is up-to-date with the Upstream branch.
> * [ ]  I have followed the standard code convention and my code contains the necessary documentation ,comments,input and output
> 
> #### Changes proposed in this pull request:
> 
> #### Languages Used:
> * [ ]  C
> * [ ]  C++
> * [ ]  Java
> * [ ]  Python
> * [ ]  JavaScript
> * [ ]  Ruby
> * [ ]  Go
> * [ ]  Machine Learning Algorithm
> * [ ]  Others,please specify  below
>   --#Mention Other Languages
> 
> #### Files Added(specify there names):
> 
> A template according to the suggestions till now ?? how is it ??


This looks great to me. But additionally, just wanted to edit the sample input, output and necessary document part by putting few more checkboxes of like...

* Neccesary Documents provided
- [ ] Sample Input/Output
- [ ] Logic Documentation
- [ ] None

Any reviews @jainaman224 ?

**Comment by helios1101** [comment]
> @helios1101 mention khushal in that feature template issue as well.
> All 3 of us try to make good templates.

Sure !

**Comment by jainaman224** [comment]
Great I am adding PR template now.
Awesome work.

**Comment by khushal87** [comment]
Thanks alot😊

**Comment by jainaman224** [comment]
I am leaving this issue open for further discussion.

**Comment by khushal87** [comment]
okay @jainaman224
